api/browser_context.go

```diff
@@ -35,4 +38,7 @@ type BrowserContext interface {
 	StorageState(opts goja.Value)
 	Unroute(url goja.Value, handler goja.Callable)
 	WaitForEvent(event string, optsOrPredicate goja.Value) any
+
+	// Internally available methods
+	Ctx() context.Context
```

Comment:
Another solution could be to have a `BrowserProcessRegister` which would keep track of all the running browser process (i.e. store the pid). When the `panic` occurs, it would just be a case of running something like `bpr.Shutdown()`.

This way we wouldn't dirty the APIs with these "internal" exported methods.

Another reason to work with a `BrowserProcessRegister` would be to clean up the context a bit by removing the the pid from the context.

Reply:
I have to understand better how this would be done, but on a first sight this seems like a better option to me. Otherwise we'll have to expose a method in order to retrieve the `ctx` from every mapping function (as a panic could be called from methods on each one of them), right?

Reply:
I've created https://github.com/grafana/xk6-browser/pull/755 which uses a register.

Reply:
Nice! I'd prefer merging #755 first and then applying those changes here then 👍